teamcity/datadog_checks/teamcity/metrics.py

```diff
@@ -0,0 +1,209 @@
+    'PassedTestCount': {'name': 'passed_test_count', 'method': 'gauge'},
+    'FailedTestCount': {'name': 'failed_test_count', 'method': 'gauge'},
+    'serverSideBuildFinishing': {'name': 'server_side_build_finishing', 'method': 'gauge'},
+    'SuccessRate': {'name': 'success_rate', 'method': 'gauge'},
+    'TimeSpentInQueue': {'name': 'time_spent_in_queue', 'method': 'gauge'},
+    'TotalTestCount': {'name': 'total_test_count', 'method': 'gauge'},
+    'VisibleArtifactsSize': {'name': 'visible_artifacts_size', 'method': 'gauge'},
+    'CodeCoverageB': {'name': 'code_coverage.blocks.pct', 'method': 'gauge'},
+    'CodeCoverageC': {'name': 'code_coverage.classes.pct', 'method': 'gauge'},
+    'CodeCoverageL': {'name': 'code_coverage.lines.pct', 'method': 'gauge'},
+    'CodeCoverageM': {'name': 'code_coverage.methods.pct', 'method': 'gauge'},
+    'CodeCoverageR': {'name': 'code_coverage.branches.pct', 'method': 'gauge'},
+    'CodeCoverageS': {'name': 'code_coverage.statements.pct', 'method': 'gauge'},
+    'CodeCoverageAbsBCovered': {'name': 'code_coverage.blocks.covered', 'method': 'gauge'},
+    'CodeCoverageAbsBTotal': {'name': 'code_coverage.blocks.total', 'method': 'gauge'},
+    'CodeCoverageAbsCCovered': {'name': 'code_coverage.classes.covered', 'method': 'gauge'},
+    'CodeCoverageAbsCTotal': {'name': 'code_coverage.classes.total', 'method': 'gauge'},
+    'CodeCoverageAbsLCovered': {'name': 'code_coverage.lines.covered', 'method': 'gauge'},
+    'CodeCoverageAbsLTotal': {'name': 'code_coverage.lines.total', 'method': 'gauge'},
+    'CodeCoverageAbsMCovered': {'name': 'code_coverage.methods.covered', 'method': 'gauge'},
+    'CodeCoverageAbsMTotal': {'name': 'code_coverage.methods.total', 'method': 'gauge'},
+    'CodeCoverageAbsRCovered': {'name': 'code_coverage.branches.covered', 'method': 'gauge'},
+    'CodeCoverageAbsRTotal': {'name': 'code_coverage.branches.total', 'method': 'gauge'},
+    'CodeCoverageAbsSCovered': {'name': 'code_coverage.statements.covered', 'method': 'gauge'},
+    'CodeCoverageAbsSTotal': {'name': 'code_coverage.statements.total', 'method': 'gauge'},
+    'DuplicatorStats': {'name': 'duplicator_stats', 'method': 'gauge'},
+    'IgnoredTestCount': {'name': 'ignored_test_count', 'method': 'gauge'},
+}
+
+REGEX_BUILD_STATS_METRICS = [
+    {
+        'regex': r'buildStageDuration\:([\s\S]*)',
+        'name': 'build_stage_duration',
+        'tags': ('build_stage',),
+        'method': 'gauge',
+    },
+    {
+        'regex': r'queueWaitReason\:([\s\S]*)',
+        'name': 'queue_wait_reason',
+        'tags': ('reason',),
+        'method': 'gauge',
+    },
+]
+
+
+def build_metric(metric_name):
+    additional_tags = []
+    name = None
+    method = None
+    if metric_name in SIMPLE_BUILD_STATS_METRICS:
+        metric_mapping = SIMPLE_BUILD_STATS_METRICS[metric_name]
+        name = metric_mapping['name']
+        method = metric_mapping['method']
```

Comment:
Nit. For clarity, since we're using this `method` as the metric type. Would it be clearer to just name the variable `metric_type`?

Reply:
makes sense